app/src/main/java/org/mozilla/fenix/session/OnRegisterActLifeCycleCallback.kt

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.session
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import org.mozilla.fenix.components.Performance
+
+/**
+ * This ActivityLifecycleCallbacks implementations tracks if a frame is visible and
+ * allows us to start a queue of tasks that have been delayed until after visual completeness.
+ */
+@SuppressWarnings("EmptyFunctionBlock")
+class OnRegisterActLifeCycleCallback(private val performance: Performance) :
+    Application.ActivityLifecycleCallbacks {
+
+    override fun onActivityStarted(activity: Activity?) {}
+
+    override fun onActivityStopped(activity: Activity?) {}
+
+    override fun onActivityResumed(activity: Activity?) {}
+
+    override fun onActivityPaused(activity: Activity?) {}
+
+    override fun onActivitySaveInstanceState(activity: Activity?, bundle: Bundle?) {}
+
+    override fun onActivityDestroyed(activity: Activity?) {}
+
+    override fun onActivityCreated(activity: Activity?, bundle: Bundle?) {
```

Comment:
nit: can we put this at the top of the class so we don't have to scroll to access it?